syntax = "proto3";

package inventory;

import "type/uuid.proto";
import "google/protobuf/timestamp.proto";

message BoundItemRequest {
    uuid id = 1;
    uint64 inbound = 2;
    uint64 outbound = 3;
    uint64 balance = 4;
    string comment = 5;
    google.protobuf.Timestamp created_at = 6;
    google.protobuf.Timestamp updated_at = 7;
}

message BoundItemResponse {
    uuid id = 1;
    uint64 inbound = 2;
    uint64 outbound = 3;
    uint64 balance = 4;
    string comment = 5;
    google.protobuf.Timestamp created_at = 6;
    google.protobuf.Timestamp updated_at = 7;
}

service BoundItem {
    rpc GetBoundItem(BoundItemRequest) returns (BoundItemResponse) {};
    rpc GetBoundItems(BoundItemRequest) returns (stream BoundItemResponse) {};
}